driver/src/orderbook/streamed/updating_orderbook.rs

```diff
@@ -7,177 +7,142 @@ use crate::{
     models::{AccountState, Order},
     orderbook::StableXOrderBookReading,
 };
-use anyhow::{anyhow, bail, ensure, Result};
+use anyhow::{anyhow, bail, Result};
 use block_timestamp_reading::{BlockTimestampReading, CachedBlockTimestampReader};
 use ethcontract::{contract::Event, BlockNumber, H256};
-use futures::{
-    channel::oneshot, compat::Future01CompatExt as _, future::FutureExt, pin_mut, select_biased,
-};
+use futures::compat::Future01CompatExt as _;
 use orderbook::Orderbook;
 use std::collections::HashSet;
-use std::sync::{
-    atomic::{AtomicBool, Ordering},
-    Arc, Mutex,
-};
-use std::{process, thread, time::Duration};
+use std::sync::{Arc, Mutex};
 
 /// An event based orderbook that automatically updates itself with new events from the contract.
-#[derive(Debug)]
 pub struct UpdatingOrderbook {
-    orderbook: Arc<Mutex<Orderbook>>,
-    // Indicates whether the background thread has caught up with past events at which point the
-    // orderbook is ready to be read.
-    orderbook_ready: Arc<AtomicBool>,
-    // When this struct is dropped this sender will be dropped which makes the updater thread stop.
-    _exit_tx: oneshot::Sender<()>,
+    contract: Arc<dyn StableXContract + Send + Sync>,
+    web3: Web3,
+    behind_mutex: Mutex<BehindMutex>,
```

Comment:
I think we may have lost some commentary here.

Reply:
Do you mean you would like there to be a comment explaining why this is needed? Sure, I'll add it.